tools/ocean_data_view_manager/tool_odv.xml

```diff
@@ -0,0 +1,180 @@
+<tool id="tool_odv" name="ODV collection manager" version="@VERSION@+galaxy@VERSION_SUFFIX@" profile="22.05" license="MIT">
+    <description>tool for qualificiation and validation</description>
+    <macros>
+        <token name="@VERSION@">0.2</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <requirements>
+        <container type="docker">pokapok/qcv-odv-tool:@VERSION@</container>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        export HOME=\$PWD &&
+        mkdir -p /runtime/config &&
+        mkdir -p '/runtime/data-in-raw/' &&
+        mkdir -p '/runtime/data-in-ref/' &&
+        mkdir -p '/runtime/data-out/' &&
+
+        cp '$input_raw' '/runtime/data-in-raw/${input_raw.element_identifier}' &&
+        #if $input_ref:            
+            cp '$input_ref' '/runtime/data-in-ref/${input_ref.element_identifier}' &&
+        #end if
+
+        
+        #if $config.type == 'yes':
+            ## Write YAML content
+            cp '${config_file}' '/runtime/config/config_user_odv_tool.yml' &&
+        #end if 
+
+        /app/launchers/start-app.sh '$mode' &&
+
+        #if $mode == 'R_APP_QV':
+            cp '/runtime/data-out/odv_collection.txt' $output &&
+        #else:
+            cp '/runtime/data-out/demo_odv_collection.txt' $output &&
+        #end if
+        cp -r '/runtime/data-out/' './'
+    ]]></command>
+    <configfiles>
+        <configfile name="config_file">
+            default:
+                file_name_list_raw: '${input_raw.element_identifier}'
+                operator: '${config.operator}' 
+            QV:
+                file_name_list_ref: '${input_ref.element_identifier}' 
+                odv_collection:
+                    name: "odv_collection.txt"
+                    qc_convention: '${config.qc_convention}' 
+                tool_option:
+                    subsetting: '${config.subsetting}' 
+                    plt: '${config.plt}' 
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param name="input_raw" type="data" format="netcdf" label="Input raw data"/>
+        <param name="input_ref" type="data" format="netcdf" optional="true" label="Input reference data"/>
+        <param name="mode" type="select" label="Select in which mode you want to run the tool">
+            <option value="R_APP_QV">QV that creates a odv collection with raw and ref files to qualify</option>
+            <option value="DEMO">Valide with ODV demo that performes the demo</option>
+        </param>
+        <conditional name="config">
+            <param name="type" type="select" label="Select if you are have your own configuration file or not.">   
+                <option value="no">No, I don't have a configuration file, I use the default one</option>
+                <option value="yes">Yes, I do have my own configuration file</option>
```

Comment:
I find this confusing. Yes, means Galaxy creates the config file? I would expect that the user uploads the file based on this text.

Reply:
Yes means you want to write your config file 
I am going to rewritte that your right it's confusing